docs/sphinx/specification/cudaq/algorithmic_primitives.rst

```diff
@@ -1,24 +1,31 @@
 Quantum Algorithmic Primitives
 ******************************
-The CUDA Quantum model provides the implementation of common quantum algorithmic
-primitives within the :code:`cudaq::` namespace. Here we enumerate available
-function calls:
+**[1]** The general philosophy of the CUDA Quantum specification is that quantum 
+device code should be encapsulated as stand-alone callable instances of generic 
+signature, and that operations or primitive algorithms targeting a quantum 
+coprocessor be implemented as adaptors on those callable instances. By adaptor, 
+we mean some generic function that takes a kernel as input with its associated 
+runtime arguments (following the signature of the kernel) and affects execution 
+of the kernel with general pre- and post-processing applied. 
```

Comment:
The last sentence is doing several heavy lifts. It might flow easier if we split it up.

1. Define adaptor, but maybe define kernel beforehand?
2. Explain that arguments to the adaptor flow through the adaptor to the kernel.
3. Explain that execution of all of this involves pre- and post-processing (abstracted away).